src/input/datetime.rs

```diff
@@ -109,6 +109,56 @@ impl<'a> EitherTimedelta<'a> {
             Self::Raw(duration) => duration_as_pytimedelta(py, duration),
         }
     }
+
+    pub fn total_seconds(&self) -> PyResult<f64> {
+        match self {
+            Self::Raw(timedelta) => {
+                let days: f64 = f64::from(timedelta.day);
+                let seconds: f64 = f64::from(timedelta.second);
+                let microseconds: f64 = f64::from(timedelta.microsecond);
+                let total_seconds: f64 = if !timedelta.positive {
+                    -1.0 * (86400.0 * days + seconds + microseconds / 1_000_000.0)
+                } else {
+                    86400.0 * days + seconds + microseconds / 1_000_000.0
+                };
+                Ok(total_seconds)
+            }
+            Self::PyExact(py_timedelta) => {
+                let days: f64 = f64::from(py_timedelta.get_days()); // -999999999 to 999999999
+                let seconds: f64 = f64::from(py_timedelta.get_seconds()); // 0 through 86399
+                let microseconds: f64 = f64::from(py_timedelta.get_microseconds()); // 0 through 999999
+                Ok(86400.0 * days + seconds + microseconds / 1_000_000.0)
+            }
+            Self::PySubclass(py_timedelta) => py_timedelta
+                .call_method0(intern!(py_timedelta.py(), "total_seconds"))?
+                .extract(),
+        }
+    }
+
+    pub fn total_milliseconds(&self) -> PyResult<f64> {
+        match self {
+            Self::Raw(timedelta) => {
+                let days: f64 = f64::from(timedelta.day);
+                let seconds: f64 = f64::from(timedelta.second);
+                let microseconds: f64 = f64::from(timedelta.microsecond);
+                let total_seconds: f64 = if !timedelta.positive {
+                    -1.0 * (86_400_000.0 * days + seconds * 1_000.0 + microseconds / 1_000.0)
+                } else {
+                    86_400_000.0 * days + seconds * 1_000.0 + microseconds / 1_000.0
+                };
+                Ok(total_seconds)
+            }
+            Self::PyExact(py_timedelta) => {
+                let days: f64 = f64::from(py_timedelta.get_days()); // -999999999 to 999999999
+                let seconds: f64 = f64::from(py_timedelta.get_seconds()); // 0 through 86399
+                let microseconds: f64 = f64::from(py_timedelta.get_microseconds()); // 0 through 999999
+                Ok(86_400_000.0 * days + seconds * 1_000.0 + microseconds / 1_000.0)
```

Comment:
I think to keep full precision, we need to try to keep full microsecond precision as integer arithmetic and do floating point cast at the last minute. If we work in `i64` then we might overflow on the conversion from days & seconds to micros for large values of `days`, but in that case the precision on the microseconds won't matter much anyway.

So I get something like this:

```suggestion
                let days: i64 = py_timedelta.get_days().into(); // -999999999 to 999999999
                let seconds: i64 = py_timedelta.get_seconds().into(); // 0 through 86399
                let microseconds = py_timedelta.get_microseconds(); // 0 through 999999

                let days_seconds = (86_400 * days) + seconds;
                if let Some(days_seconds_as_micros) = days_seconds.checked_mul(1_000_000) {
                    let total_microseconds = days_seconds_as_micros + i64::from(microseconds);
                    Ok(total_microseconds as f64 / 1_000.0)
                } else {
                    // Fall back to floating-point operations if the multiplication overflows
                    let total_milliseconds = days_seconds as f64 * 1_000.0 + f64::from(microseconds) / 1_000.0;
                    Ok(total_milliseconds)
                }
```

... and I guess we can do similar for the other cases.